React: Merge "Using NodeJS For Your Backend" lesson with "Conclusion" (#26278)
TheOdinProject/curriculum: react/conclusion/conclusion_full_stack_javascript.md.md

Furthermore, you can dive into React meta frameworks. They're a framework for React that add first-class support for functionalities like data fetching and routing. Additionally, they leverage React's latest features to provide the best experience possible for developers and users alike. Some examples of these are [Next](https://nextjs.org/) and [Remix](https://remix.run/).

You're at a point in your React journey where you can begin learning about design patterns and architecture. [patterns.dev](https://www.patterns.dev/) is an exceptional resource that will help you build better React apps by leveraging powerful patterns. They are worth a bookmark!

Before you move onto the next section. Fill out this [React course feedback survey](https://docs.google.com/forms/d/e/1FAIpQLSdj_tNMp0LEz3ZLPqYcF67V11tX_CCJP3CTictPZzZ6XQm2Gw/viewform?usp=sf_link) to add your input and experience with the section. This helps us improve the section as well as the overall course and is greatly appreciated.

> ### Using NodeJS for your backend
> 
> So, you've now learned a lot of juicy JavaScript, enabling you to make various components of your HTML/CSS page react to user actions and/or input. That's great! As you've seen, you can get your apps to do a lot of pretty cool things already, just using client-side JavaScript. You may have noticed, though, that there is still an important piece of the puzzle missing: unless you use [Local Storage](http://coding.smashingmagazine.com/2010/10/11/local-storage-and-how-to-use-it/), your app 'forgets' the user's preferences, as well as any other changes made, as soon as the page gets reloaded.
> 
> While Local Storage is great, it's not ideal: it only stores data on the computer from which the user is accessing the page. This approach does not allow for the app to 'remember' anything when the same user tries to access it again from a different device. For that, you're going to need a real backend.
> 
> ### NodeJS to the rescue!
> 
> We are now ready to leave the frontend to learn all about how to store persistent data.
> NodeJS is a JavaScript runtime environment which will let you build an entire backend with JavaScript. We'll learn all about it next in our [NodeJS](https://www.theodinproject.com/paths/full-stack-javascript/courses/nodejs) course. With Node, we'll be able to add a bunch of cool features to our React apps like user authentication, data persistence, and more.

Reviewer: I feel like these two sections can be combined into one + some other tweaks:

```suggestion
### Using a backend

As you've seen, you can get your apps to do a lot of pretty cool things already, just using client-side JavaScript. You may have noticed, though, that there is still an important piece of the puzzle missing: unless you use [Local Storage](http://coding.smashingmagazine.com/2010/10/11/local-storage-and-how-to-use-it/), your app 'forgets' the user's preferences, as well as any other changes made, as soon as the page gets reloaded.

While Local Storage is great, it's not ideal: it only stores data on the computer from which the user is accessing the page. This approach does not allow for the app to 'remember' anything when the same user tries to access it again from a different device. For that, you're going to need a real backend, which you'll learn all about next in our [NodeJS](https://www.theodinproject.com/paths/full-stack-javascript/courses/nodejs) course. With Node, we'll be able to add a bunch of cool features to your apps like user authentication, data persistence, and more.
```
